java/flight/flight-core/src/main/java/org/apache/arrow/flight/ArrowMessage.java

```diff
@@ -137,21 +156,24 @@ public ArrowMessage(FlightDescriptor descriptor, Schema schema, IpcOption option
     this.descriptor = descriptor;
     this.appMetadata = null;
     this.bodyCompression = NoCompressionCodec.DEFAULT_BODY_COMPRESSION;
+    this.tryZeroCopyWrite = false;
   }
 
   /**
    * Create an ArrowMessage from a record batch and app metadata.
    * @param batch The record batch.
    * @param appMetadata The app metadata. May be null. Takes ownership of the buffer otherwise.
+   * @param tryZeroCopy Whether to enable the zero-copy optimization.
    */
-  public ArrowMessage(ArrowRecordBatch batch, ArrowBuf appMetadata, IpcOption option) {
+  public ArrowMessage(ArrowRecordBatch batch, ArrowBuf appMetadata, boolean tryZeroCopy, IpcOption option) {
```

Comment:
is there a reason to not reference the static variable directly instead of pass this through?  testability?

Reply:
For testability, and so that it can be toggled at runtime/on a per-RPC basis instead of only globally/statically.